src/main/java/com/google/api/generator/gapic/composer/SampleCodeHelperComposer.java

```diff
@@ -124,6 +158,41 @@ private static AssignmentExpr assignClientVariableWithCreateMethodExpr(
         .build();
   }
 
+  private static Expr assignMethodArgumentWithDefaultValue(
+      MethodArgument argument, Map<String, ResourceName> resourceNames) {
+    return AssignmentExpr.builder()
+        .setVariableExpr(
+            createVariableDeclExpr(JavaStyle.toLowerCamelCase(argument.name()), argument.type()))
+        .setValueExpr(DefaultValueComposer.createDefaultValue(argument, resourceNames))
+        .build();
+  }
+
+  private static Expr createAssignExprForVariableWithClientMethod(
+      String variableName,
+      TypeNode variableType,
+      VariableExpr clientVarExpr,
+      String methodName,
+      List<MethodArgument> arguments) {
+    VariableExpr varExpr = createVariableExpr(variableName, variableType);
+    MethodInvocationExpr clientMethodInvocationExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(clientVarExpr)
+            .setMethodName(JavaStyle.toLowerCamelCase(methodName))
+            .setArguments(mapMethodArgumentsToVariableExprs(arguments))
+            .setReturnType(variableType)
+            .build();
+    return AssignmentExpr.builder()
+        .setVariableExpr(varExpr.toBuilder().setIsDecl(true).build())
+        .setValueExpr(clientMethodInvocationExpr)
+        .build();
+  }
+
+  private static List<Expr> mapMethodArgumentsToVariableExprs(List<MethodArgument> arguments) {
+    return arguments.stream()
```

Comment:
As previously mentioned in some reviews, this is another instance where we're creating the `VariableExpr` twice - once for the assignment of the default value, and another time for the method argument. Could we please refactor this such that we only create them once?

One way is to create the set of variables in the original method, then pass the variable into `DefaultValueComposer`. Also, we can add a `VariableExpr` variant to `DefaultValueComposer` as well.

Reply:
Very useful. Updated the code.